crates/autopilot/src/auction_transaction.rs

```diff
@@ -30,10 +30,13 @@ pub struct AuctionTransactionUpdater {
 }
 
 impl AuctionTransactionUpdater {
-    // TODO: spawn this in autopilot
     pub async fn run_forever(self) -> ! {
         loop {
-            match self.update().await {
+            match self
+                .update()
+                .instrument(tracing::info_span!("auction_transaction_update"))
```

Comment:
nit: Can we instrument the entire `run_forever()` call?

Reply:
Done. Moved the instrument into lib.rs.